delira/training/backends/torch/experiment.py

```diff
@@ -8,103 +7,240 @@
         checkpoint_freq : int
-            frequency of saving checkpoints (1 denotes saving every epoch,
-            2 denotes saving every second epoch etc.); default: 1
-        trainer_cls : subclass of :class:`PyTorchNetworkTrainer`
-            the trainer class to use for training the model, defaults to
-            :class:`PyTorchNetworkTrainer`
+            determines how often to checkpoint the training
+        reduce_mode : str
+            determines how to reduce metrics; must be one of
+            'mean' | 'sum' | 'first_only'
+        val_score_key : str
+            specifies which metric to use for best model selection.
+        val_score_mode : str
+            determines whether a high or a low val_score is best. Must be one
+            of 'highest' | 'lowest'
+        val_freq : int
+            specifies how often to run a validation step
+        callbacks : list
+            list of callbacks to use during training. Each callback should be
+            derived from :class:`delira.training.callbacks.AbstractCallback`
+        trainer_cls : type
+            the class implementing the actual training routine
         **kwargs :
             additional keyword arguments
 
+        Returns
+        -------
+        :class:`AbstractPyTorchNetwork`
+            the trained model
+
         """
 
-        if key_mapping is None:
-            key_mapping = {"x": "data"}
-        super().__init__(params=params, model_cls=model_cls,
-                         n_epochs=n_epochs, name=name, save_path=save_path,
-                         key_mapping=key_mapping,
-                         val_score_key=val_score_key,
-                         optim_builder=optim_builder,
-                         checkpoint_freq=checkpoint_freq,
-                         trainer_cls=trainer_cls,
-                         **kwargs)
-
-    def kfold(self, data: BaseDataManager, metrics: dict, num_epochs=None,
-              num_splits=None, shuffle=False, random_seed=None,
-              split_type="random", val_split=0.2, label_key="label",
-              train_kwargs: dict = None, test_kwargs: dict = None,
-              metric_keys: dict = None, params=None, verbose=False,
-              **kwargs):
+        return super().run(
+            params=params, train_data=train_data, val_data=val_data,
+            optim_builder=optim_builder, gpu_ids=gpu_ids,
+            checkpoint_freq=checkpoint_freq, reduce_mode=reduce_mode,
+            val_score_key=val_score_key, val_score_mode=val_score_mode,
+            val_freq=val_freq, callbacks=callbacks,
+            trainer_cls=trainer_cls, **kwargs)
+
+    def test(self, model, test_data, prepare_batch, callbacks=None,
+             predictor_cls=Predictor, metrics=None, metric_keys=None,
+             **kwargs):
+        """
+        Setup and run testing on a given network
+        Parameters
+        ----------
+        model : :class:`AbstractNetwork`
```

Comment:
It would be great if the order of the argument descriptions is the same as the order in the call (make life a little bit easier ^^ )